prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -27,6 +32,50 @@ import kotlin.reflect.jvm.isAccessible
  * A collection of helper functions that modify an existing FHIR bundle.
  */
 const val conditionExtensionurl = "https://reportstream.cdc.gov/fhir/StructureDefinition/reportable-condition"
+const val conditionCodeExtensionURL = "https://reportstream.cdc.gov/fhir/StructureDefinition/condition-code"
+
+private fun lookupCondition(code: Coding, metadata: Metadata): Coding? {
+    val mappingTable = metadata.findLookupTable("observation-mapping").also {
+        if (it == null) {
+            return null // TODO: better error if table is missing? log vs throw?
+        }
+    }!!
+    val condition = mappingTable.dataRowsMap.find {
+        it[ObservationMappingConstants.TEST_CODE_KEY.lowercase()] == code.code // TODO: why lower
+    }
+    return if (condition.isNullOrEmpty()) {
+        null
+    } else {
+        Coding(
+            condition[ObservationMappingConstants.CONDITION_CODE_SYSTEM_KEY.lowercase()], // TODO: why lower
+            condition[ObservationMappingConstants.CONDITION_CODE_KEY.lowercase()], // TODO: why lower
+            condition[ObservationMappingConstants.CONDITION_NAME_KEY.lowercase()] // TODO: why lower
+        )
+    }
+}
+
+fun Observation.addMappedCondition(metadata: Metadata): List<ActionLogDetail> {
+    val codeSourcesMap = mapOf(
+        ObservationMappingConstants.BUNDLE_CODE_IDENTIFIER to this.code.coding,
+        ObservationMappingConstants.BUNDLE_CODEABLE_IDENTIFIER to this.valueCodeableConcept.coding
+    ).filterValues { it.isNotEmpty() }
+    if (codeSourcesMap.values.flatten().isEmpty()) return listOf(UnmappableConditionMessage())
+
+    return codeSourcesMap.mapNotNull { codeSourceEntry ->
+        codeSourceEntry.value.mapNotNull { code ->
+            lookupCondition(code, metadata).let { conditionCode ->
+                if (conditionCode == null) {
+                    code.code
+                } else {
+                    code.addExtension(conditionCodeExtensionURL, conditionCode)
+                    null
+                }
+            }
+        }.let {
+            if (it.isEmpty()) null else UnmappableConditionMessage(it, codeSourceEntry.key)
```

Comment:
Is this logic potentially not granular enough?  If there was FHIR like

```json
"coding" : [
  {
    "code":  123 // Can be mapped
  },
  {
    "code"  456 // Cannot be mapped
  }
]
``` 
We would not return the error log?  Maybe I'm misreading this.

Either way it would be nice for the error message to also indicate which `code` was not ampped

Reply:
Should behave as you expect/describ:

We would return the error log for the unmapped code in the 'chimera' FHIR, to include the code in the message and the origin field in the fieldMapping value